Packet++/header/TcpLayer.h

```diff
@@ -236,7 +236,7 @@ namespace pcpp
 			if (tlvDataLen < sizeof(TLVRawData::recordType))
 				return false;
 
-			if (data->recordType == (uint8_t)PCPP_TCPOPT_NOP || data->recordType == (uint8_t)PCPP_TCPOPT_EOL)
+			if (data->recordType == (uint8_t)TcpOptionType::PCPP_TCPOPT_NOP || data->recordType == (uint8_t)TcpOptionType::PCPP_TCPOPT_EOL)
```

Comment:
since we are now doing refactoring, maybe it's time to get rid of C style casting. In addition, I feel like here is better to have a convert function, such as `GetTcpOptionTypeFromRecordType(data->recordType) == TcpOptionType::PCPP_TCPOPT_NOP`